mautic/mautic
# Updating to Symfony 5 with Rector

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes https://mautic.atlassian.net/browse/TPROD-299, https://mautic.atlassian.net/browse/TPROD-276, https://mautic.atlassian.net/browse/TPROD-275



#### Description:

:warning: this PR is based on https://github.com/mautic/mautic/pull/11452 and that one should be merged first.

Adding Rector rules for Symfony upgrades so it does some work for us (humans) and also checks that no deprecated code is merged with new PRs.

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. This is refactoring of the whole app, so everything is affected. Good thing is that this is automatic update used by many companies so the bugs in this process should be already fixed. I would leave overall testing of the whole app for release of Mautic 5 - alpha. But the more testing you can do the better. Too bad I can't tell you what to focus on.
